src/device.c

```diff
@@ -729,19 +735,18 @@ static void on_modbus_disconnected(void *user_data)
 
 static void on_publish_data(void *data, void *user_data)
 {
-	struct knot_data_item data_item;
+	struct knot_data_item *data_item;
 	int *sensor_id = data;
 	int rc;
 
-	if (*sensor_id >= thing.data_item_count)
+	data_item = l_hashmap_lookup(thing.data_items, sensor_id);
```

Comment:
Why is the macro `L_INT_TO_PRT` needed on line 165 and 490, but not here?

Reply:
because sensor_id is a prt already.